### PR TITLE
feat: [CDS-45547]: Add base terragrunt tasks and support methods

### PR DESCRIPTION
**Pull request by abosii-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/40116)

### CODE REVIEW

File: 1 [hunk 1]
```diff
@@ -0,0 +1,27 @@
+pick 314858df3da feat: [CDS-45547]: Add base terragrunt tasks and support methods
```
Comment (vlica-harness) [inline 1:1]: This file is not needed

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/terragrunt/request/AbstractTerragruntTaskParameters.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+@FieldDefaults(level = AccessLevel.PRIVATE)
+public abstract class AbstractTerragruntTaskParameters
+    implements ExecutionCapabilityDemander, NestedAnnotationResolver {
+  @NonNull String accountId;
+  @NonNull String entityId;
+  @NonNull TerragruntRunConfiguration runConfiguration;
+  @NonNull StoreDelegateConfig configFilesStore;
+  StoreDelegateConfig backendFilesStore;
+  List<StoreDelegateConfig> varFiles;
+
```
Comment (vlica-harness) [inline 950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/terragrunt/request/AbstractTerragruntTaskParameters.java:46]: Shouldn't be targets part of this AbstractTerragruntTaskParameters?

File: 1 (hunk 1 above)
Comment (abosii-harness) [reply 1]: Done

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/terragrunt/request/AbstractTerragruntTaskParameters.java (hunk 2 above)
Comment (abosii-harness) [reply 950-delegate-tasks-beans/src/main/java/io/harness/delegate/beans/terragrunt/request/AbstractTerragruntTaskParameters.java:46]: Added

**Review by vlica-harness — COMMENTED** [review]
> Reviewed 36 of 39 files at r1, 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40116)* status: 2. unresolved discussions. 0 of 1 approvals obtained.

**Review by vlica-harness — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40116#-:-NH59Ecy1pbuwIzZdUAs:bnfp4nl)* status: 2. unresolved discussions.

**Review by tmacari-harness — DISMISSED** [review]
> Reviewed 36 of 39 files at r1, 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40116)* status: 2. unresolved discussions.

**Comment by vlica-harness** [comment]
trigger messagecheck

**Review by anilchowdhury-harness — APPROVED** [review]
:lgtm:

> Reviewed 35 of 39 files at r1, 2 of 3 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40116#-:-NHAcckT-JJcDs20U3dN:bnfp4nl)* status: 2. unresolved discussions.

**Review by vlica-harness — APPROVED** [review]
> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40116)* status: 2. unresolved discussions.

**Review by vlica-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40116)* status: :shipit: complete! ✓

**Pull request merged**